### PR TITLE
Add offer grouping and sorting to the NGS table view

### PR DESCRIPTION
**Pull request by andrejak**
This PR adds an offer grouping function and column sorting functionality to the NGS table view, with some slightly better examples. We'll need to check a couple of things with the back-end team once the API is finalised but I think it's mostly there. 

A couple UI questions to confirm - the font weight is different for the highlighted columns but is the jump too much visually? 
Here's a video of the highlighting and sorting, let me know if anything doesn't behave as expected:

https://github.com/duffelhq/duffel-components/assets/1416759/815f67bb-e384-48bc-894f-ce6d884b27fa

### CODE REVIEW

File: src/components/DuffelNGSView/DuffelNGSView.tsx [hunk 1]
```diff
@@ -26,40 +26,59 @@ export const DuffelNGSView: React.FC<DuffelNGSViewProps> = ({
...
+  React.useEffect(() => {
+    if (sortSettings.shelf) {
+      const sortedRows = [...rows].sort((a, b) => {
+        const aAmount = +(a[sortSettings.shelf!]?.total_amount || 0);
+        const bAmount = +(b[sortSettings.shelf!]?.total_amount || 0);
+        if (aAmount && bAmount) {
+          return sortSettings.direction === "asc"
+            ? aAmount - bAmount
+            : bAmount - aAmount;
+        }
```
Comment (igorp1) [inline src/components/DuffelNGSView/DuffelNGSView.tsx:46]: How about we extract this into a helper? Just to keep the component leaner and a bit easier to read through?

File: src/components/DuffelNGSView/DuffelNGSView.tsx [hunk 2]
```diff
@@ -26,40 +26,59 @@ export const DuffelNGSView: React.FC<DuffelNGSViewProps> = ({
...
-              <th key={shelf} onClick={() => setSelectedColumn(shelf)}>
+              <th
+                key={shelf}
+                onClick={() => {
+                  if (shelf === sortSettings.shelf) {
+                    setSortSettings({
+                      shelf,
+                      direction:
+                        sortSettings.direction === "asc" ? "desc" : "asc",
+                    });
```
Comment (igorp1) [inline src/components/DuffelNGSView/DuffelNGSView.tsx:72]: How about we separate the direction from the sort type? That way we could simplify this logic a bit into something like `toggleSortDirection`

File: src/components/DuffelNGSView/lib/group-offers-for-ngs-view.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+export const groupOffersForNGSView = (
+  offers: OfferWithNGS[],
+  sliceIndex: number
+): NGSOfferRow[] => {
+  const offersMap: Record<string, NGSOfferRow> = {};
+  offers.forEach((offer) => {
+    if (sliceIndex > offer.slices.length) {
+      throw new Error("Invalid slice index");
```
Comment (igorp1) [inline src/components/DuffelNGSView/lib/group-offers-for-ngs-view.ts:20]: For more clarity on the error, how about: 
```suggestion
      throw new Error("Attempted to call `groupOffersForNGSView` with an Invalid slice index");
```

**Review by igorp1 — APPROVED** [review]
Left some suggestions but non blocking! Looks great!

File: src/components/DuffelNGSView/DuffelNGSView.tsx (hunk 2 above)
Comment (andrejak) [reply src/components/DuffelNGSView/DuffelNGSView.tsx]: yeah I was debating that when I set up the sort settings... Makes sense, done!

**Comment by andrejak** [comment]
I'm going to merge this for now to avoid stacking PRs but if there's any further feedback please just let me know and I'll address it in a follow up PR :)

**Comment by srcltd** [comment]
LGTM 🙌🏼

**Pull request merged**